### PR TITLE
TRUNK-5896: VisitAttributeType Domain - Switching from Hibernate Mappings to Annotations

### PR DESCRIPTION
**Pull request by alinmihaila99**
## Issue I worked on
see https://issues.openmrs.org/browse/TRUNK-5896

## Checklist: I completed these to help reviewers :)
- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.
- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)
- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] All new and existing **tests passed**.
- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: api/src/main/java/org/openmrs/VisitAttributeType.java [hunk 1]
```diff
@@ -12,12 +12,24 @@
...
  * A user-defined extension to the {@link Visit} class.
  * @see AttributeType 
  */
+@Entity
+@Table(name = "visit_attribute_type")
 public class VisitAttributeType extends BaseAttributeType<Visit> implements AttributeType<Visit> {
-	
+
+	@Id
+	@Column(name = "visit_attribute_type_id")
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/VisitAttributeType.java:31]: looks good cc @ibacher  @Ruhanga

**Comment by gitcliff** [comment]
@Ruhanga  we can merge this ..thanks

**Comment by sherrif10** [comment]
LGTM  i think this is the best xperience with hibernate  thanks everyone have liked though not commenting  but following

**Review by Ruhanga — APPROVED** [review]

**Comment by Ruhanga** [comment]
@alinmihaila99, could you rebase this to resolve merge conflicts?

**Review by Ruhanga — APPROVED** [review]

**Pull request closed without merging**